### PR TITLE
Create a command line API

### PR DESCRIPTION
**Pull request by Honny1**
This PR creates a command line API.

### CODE REVIEW

File: oscap_report/cli.py [hunk 1]
```diff
@@ -1,8 +1,133 @@
...
+    def generate_report(self, report_parser):
+        logging.info("Generate report")
+        report_generator = ReportGenerator(report_parser)
+        if self.output_format == "HTML":
+            return report_generator.generate_html_report()
+        return None
+
+    def load_file(self):
+        logging.info("Loading file")
+        logging.debug(self.report_file)
```
Comment (evgenyz) [inline oscap_report/cli.py:107]: It's good to pair run-time values with static text. In that case searching in source code for `Loading file` would bring you right where it was emitted. Also I see no sense in having these two messages on different level.

Like `logging.info("Loading file: %s", self.report_file)`

File: oscap_report/cli.py [hunk 2]
```diff
@@ -1,8 +1,133 @@
...
+        return parser.parse_args()
+
+    @staticmethod
+    def _prepare_arguments(parser):
+        parser.add_argument(
+            'FILE',
+            type=argparse.FileType("r"),
+            nargs='?',
+            default=stdin,
+            help="ARF file, stdin if not provided")
```
[the remark this answers is not in the document]
Comment (jan-cerny) [reply oscap_report/cli.py]: :zany_face:

File: oscap_report/cli.py [hunk 3]
```diff
@@ -1,8 +1,137 @@
...
+
 from . import __version__
+from .old_html_report_style.report_generator import ReportGenerator
+from .report_parser.report_parser import ReportParser
+
+DESCRIPTION = ("Generate a HTML (JSON, PDF?, Printable HTML, etc) document (HTML report)"
+               " from an ARF (or XCCDF file) containing results of oscap scan. Unless"
+               " the --output option is specified it will be written to standard output.")
+LOG_LEVES_DESCRIPTION = (
+    "LOG LEVES:\n"
```
Comment (jan-cerny) [inline oscap_report/cli.py:13]:
```suggestion
    "LOG LEVELS:\n"
```

File: oscap_report/report_parser/report_parser.py [hunk 4]
```diff
@@ -0,0 +1,4 @@
+
+class ReportParser():
```
Comment (jan-cerny) [inline oscap_report/report_parser/report_parser.py:2]: Isn't ReportParser a confusing name? I'm aware that in SCAP terminology the word "report" is used for also in ARF ( the R in ARF stands for "reporting"). But when I think of a "report", the first thing that comes to my mind is the HTML report. But you want to parse ARF, or maybe XCCDF results files here. So what about renaming it to "ResultsParser" or "SCAPResultsParser" or similar? Proper naming is difficult.

**Comment by Honny1** [comment]
@evgenyz and @jan-cerny I made the changes based on your comments. Which of you will merge PR?

**Pull request merged**